### PR TITLE
Add job suspend resume paths

### PR DESCRIPTION
**Pull request by scrappywyrm**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json [hunk 1]
```diff
@@ -625,6 +625,118 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Job suspended successfully."
+          },
+          "202": {
+            "description": "Suspend operation is in progress.",
+            "headers": {
+              "Location": {
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:718]: We should also include `Azure-AsyncOperation`. This helps portal track operation status and error reason

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json [hunk 2]
```diff
@@ -794,6 +906,20 @@
                 "modelAsString": true
               }
             },
+            "runningState": {
+              "description": "Current running state of the job",
+              "enum": [
+                "Running",
+                "InProgress",
```
[the remark this answers is not in the document]
Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:913]: Synced with Anthony and Vini and the new states are: Active, InProgress and Suspended

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json [hunk 3]
```diff
@@ -625,6 +625,124 @@
...
+            "in": "path",
+            "description": "Name of the Job.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Job resumed successfully."
```
Comment (raosuhas) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:657]: [ARMBlockingComment] Your examples are returning an object , but the response here is not. Can you please fix either one ?

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json [hunk 4]
```diff
@@ -625,6 +625,124 @@
...
+        "operationId": "Jobs_Resume",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
```
Comment (raosuhas) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:644]: >/v3 [](http://example.com/codeflow?start=68&length=3)

use the v5 please

**Comment by raosuhas** [comment]
You have similar flagged errors in your LintDiff validations here : https://github.com/Azure/azure-rest-api-specs/pull/29788/checks?check_run_id=27473684037

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json (hunk 3 above)
Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json]: fixed by adding schema reference to 200 response

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json (hunk 4 above)
Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json]: done, thank you!

Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:644]: hi @raosuhas ...for v5/, i tried updating the whole `jobs.json` file but got an error that the latest stable version conflicts with the intended changes

the run with the error: https://github.com/Azure/azure-rest-api-specs/pull/29788/checks?check_run_id=27634707415

i reverted back to v3 for all versions

**Comment by bgashirabake** [comment]
cc: @raosuhas hi, could you please take a look at the recent changes? thank you!!

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (jijohn14) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:925]: Does Active mean it is not in suspended state but idle?

**Review by jijohn14 — CHANGES_REQUESTED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:925]: Active could mean idle but also could mean running. idle if job is stopped running if job is not stopped (so executions will fire).

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/Jobs_Resume.json [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    "200": {
+      "headers": {},
+      "body": {
+        "id": "/subscriptions/34adfa4f-cedf-4dc0-ba29-b6d1a69ab345/resourceGroups/rg/providers/Microsoft.App/jobs/testcontainerAppsJob0",
+        "name": "testcontainerAppsJob0",
+        "type": "Microsoft.App/jobs",
+        "location": "East US",
+        "properties": {
+          "provisioningState": "Succeeded",
+          "runningState": "Running",
```
[the remark this answers is not in the document]
Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/Jobs_Resume.json:18]: on it, thank you!!

**Comment by bgashirabake** [comment]
> i think overall looks good, maybe also update the history.rst file and also add tests

thanks @smzeng ....history.rst and tests are for cli PRs i believe?

**Comment by smzeng** [comment]
> > i think overall looks good, maybe also update the history.rst file and also add tests
> 
> thanks @smzeng ....history.rst and tests are for cli PRs i believe?

youre totally right, thats my bad haha. No wonder i couldn't find the file when i tried to link it earlier. Resolving my comments :)

**Review by jijohn14 — APPROVED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json [hunk 6]
```diff
@@ -625,6 +625,130 @@
...
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "jobName",
+            "in": "path",
+            "description": "Name of the Job.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
```
Comment (p-bouchon) [inline specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:714]: What does this regex do? I don't see it used for other jobName parameters

Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:714]: under the Parameter section, i see the same regex is used for jobName? this is at line 761 in the new edited file

Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:714]: oh...i see other apis just reference the Parameter...is that preferable?

Comment (p-bouchon) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:714]: Right I think the same parameter should be referenced

Comment (bgashirabake) [reply specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/Jobs.json:714]: got it got it

**Review by p-bouchon — APPROVED** [review]

**Comment by jijohn14** [comment]
Change will be added to next preview api-version

**Pull request closed without merging**